### DISCUSSION TITLE
[기능 요청] 본문 내 빈 줄 사라짐 현상

### DISCUSSION THREAD

**Discussion by TetraTheta** [pronist/hELLO#78] [Q&A] [answered]
# 환경

- 운영체제: Windows 11
- 브라우저: Edge (Chromium)
- 버전: 94.0.992.50
- 스킨 버전: 3.4.4

# 설명
스킨 자체의 버그라기보다는 티스토리의 강제적인 정책 때문인 것으로 추측됩니다만, 본문 내의 빈 줄의 스타일에 전부 `display: none !important`가 추가됩니다.
```html
<p data-ke-size="size16" style="display: none !important;">&nbsp;</p>
```
이렇게요. 문제는 이걸 끌 수 있는 기능을 티스토리가 제공하고 있지 않기에, 빈 줄을 다시 보여주기 위해서는 스킨의 힘을 빌어야 하는 상황입니다.

현재 최신 버전인 3.4.5보다 버전이 낮지만, 커밋 기록을 보았을 때 3.4.5에서 이 문제가 해결된 것 같지 않아보입니다.

# 발생 과정
글을 작성 시 빈 줄을 일반 줄 사이사이에 넣은 후, 글을 발행합니다. 발행된 글에서 빈줄이 사라진 것을 확인합니다.

# 스크린샷
![real](https://user-images.githubusercontent.com/37061801/137757360-636224d8-6030-4e72-a6e2-1af68883595b.png)

![f12](https://user-images.githubusercontent.com/37061801/137757379-f630cf38-9bac-41d4-8d2f-9892ab8abc92.png)

![source](https://user-images.githubusercontent.com/37061801/137757526-df2c8939-517c-4a03-83b9-1bd5fb3c1ed5.png)

각각 실제 표시되는 모습, F12 개발자 모드로 본 HTML 코드, 페이지 소스 보기로 본 본문 내용입니다.

**Comment by pronist** [comment]
@TetraTheta 흠, 버그 상황을 재현하고 싶은데 글을 작성하실 때 어떤 방법으로 하시는지 정확히 알 수가 없어 의문입니다. 타 페이지의 글을 티스토리 에디터에 **복사/붙여넣기** 하지 않는 한, 그런 증상이 발생하는 일은 많이 없는 것으로 알고 있습니다. 타 페이지의 글을 에디터에 복사/붙여넣기하게 되면 티스토리 에디터에서 자체적으로 처리하기 때문에 경우의 수가 많아 짐작하기 어려운 경우가 많답니다.

또한, **티스토리 에디터에서 동작하는 부분을 스킨에서 자체적으로 통제하는 패치는 추구하고자 하는 방향이 아닙니다.** 다른 스킨과의 호환성에도 위배 될 가능성이 있어 명확한 버그가 아니라면 패치가 진행되지 않을 것입니다.

**Comment by TetraTheta** [comment]
문제가 생긴 글 예시) https://tetralog.tistory.com/837

드래그&드랍으로 사진을 업로드 한 것을 제외하면 복사/붙여넣기라고 할 만한 행위를 한 적이 없습니다. 모든 글 내용은 웹브라우저 상에서 직접 작성된 내용입니다.

또한 `display: none !important`를 붙이는 것은 티스토리 에디터가 아니라 티스토리가 스킨과 함께 불러오는 자바스크립트로 보입니다. 그렇지 않고서야는 페이지 소스에서는 display 속성이 없는데 사용자에게 보여지는 결과물인 F12에는 붙어 나올리가 없을테니까요.

티스토리가 불러오는 자바스크립트들 중 하나에서 의심되는 구문이 몇 보이기는 했습니다만, 난독화가 되어 있어서 확신이 들지는 않네요.
의심되는 스크립트 경로: https://t1.daumcdn.net/tistory_admin/assets/blog/tistory-ef60c246ebd435f37577f30fe035738321a05d52/blogs/script/tiara/tiara.min.js?_version_=tistory-ef60c246ebd435f37577f30fe035738321a05d52

이럴 경우 '티스토리 에디터'의 동작이 아닌, '티스토리'의 동작이 되는 경우니까 가능하지 않을까 합니다. 아니라고 생각되신다면 그냥 이 이슈를 닫으셔도 됩니다.

**Comment by pronist** [comment]
@TetraTheta 혹시, 글을 수정하셨나요? 왠지 모르겠지만, 
제게는 스크린샷처럼 나타나질 않네요... Windows 10, Edge, 크롬, 웨일 브라우저에서 확인이 완료되었습니다. 

![1](https://user-images.githubusercontent.com/25624965/137915750-f4649965-043d-410d-aa60-105e7c2ad9a8.PNG)

**Comment by TetraTheta** [comment, marked as answer]
음, 스크린샷 올려주신 걸 확인해보았는데, 제 브라우저쪽 문제였던 것 같습니다. AdGuard를 끄고 페이지를 새로고침하니까 저도 줄 바꿈이 다시 돌아와 있네요.

익스텐션을 생각해보았어야 했는데, 제 불찰로 괜한 불편함을 드려 죄송합니다.
